crates/wasmparser/src/validator/types.rs

```diff
@@ -2708,6 +2708,10 @@ impl TypeList {
             | (HT::Struct, _)
             | (HT::Array, _)
             | (HT::I31, _) => false,
+
+            // TODO: this probably isn't right, this is probably related to some
+            // gc type.
+            (HT::Exn, _) => false,
```

Comment:
I expect either there will be a `NoExn` bottom type and this will be a fourth type hierarchy or this will slot under `Any`, but until those wrinkles get worked out, this is fine.

Reply:
My understanding of the longer term plan is for it be a fourth type hierarchy. Having it under `any` adds some difficulties for AOT translators like wasm2c which want it to have a different representation than any. There would also be some difficulties with JS thrown non-object values if under `any`.